### ISSUE TITLE
Random Error while waiting for package deployed

### ISSUE THREAD

**Issue by talset** [magento/magento2#21852]
### Preconditions (*)

1. Magento version: `2.2.8`

2. Magento source code is build by concourse (https://concourse-ci.org/).
The concourse task use a docker image based on php:7.1 with composer and few tools added.

3. "Build" steps:
```
composer install -o -q --ignore-platform-reqs
php -d memory_limit=2048M bin/magento setup:di:compile
rm -Rf var/composer_home var/log var/page_cache var/session
rm -Rf 'pub/static/adminhtml/*'
rm -Rf 'pub/static/frontend/*'
rm -Rf 'var/view_preprocessed/pub/*'
CONFIG__DEFAULT__DEV__JS__MINIFY_FILES=1
MAGE_MODE=production
php -d memory_limit=2048M bin/magento setup:static-content:deploy -f --exclude-theme=Magento/backend --jobs=4 fr_FR
```

### Steps to reproduce (*)

1. The issue seems quite random but ot happen with the following command `php -d memory_limit=2048M bin/magento setup:static-content:deploy -f --exclude-theme=Magento/backend --jobs=4 fr_FR`

### Expected result (*)
1. Not error returned by `magento setup:static-content:deploy`

### Actual result (*)
1.

 ```
php -d memory_limit=2048M bin/magento setup:static-content:deploy -f --exclude-theme=Magento/backend --jobs=4 fr_FR

Deploy using quick strategy
frontend/Magento/blank/fr_FR            2986/2986           ============================ 100% %  36 secs             
frontend/Magento/luma/fr_FR             3002/3002           ============================ 100% %  36 secs             
...
                                                           
  [RuntimeException]                                       
  Error while waiting for package deployed: 67; Status: 0  

setup:static-content:deploy [-f|--force] [-s|--strategy [STRATEGY]] [-a|--area [AREA]] [--exclude-area [EXCLUDE-AREA]] [-t|--theme [THEME]] [--exclude-theme [EXCLUDE-THEME]] [-l|--language [LANGUAGE]] [--exclude-language [EXCLUDE-LANGUAGE]] [-j|--jobs [JOBS]] [--symlink-locale] [--content-version CONTENT-VERSION] [--refresh-content-version-only] [--no-javascript] [--no-css] [--no-less] [--no-images] [--no-fonts] [--no-html] [--no-misc] [--no-html-minify] [--] [<languages>]...
```

The issue is really random, we have 3 environments (dev/staging/prod) and it happen randomly once a day or less. After the issue if we run the same build pipeline it works. 

Do you have any idea of what does it means ? What could be the issue ?

**Comment by magento-engcom-team** [comment]
Hi @talset. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@talset do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @talset thank you for you report, but i'm not able to reproduce this on my local environment, your error comes from ->
![DeepinScreenshot_select-area_20190321120152](https://user-images.githubusercontent.com/40756439/54745007-273bf680-4bd1-11e9-9c33-f493ae9a09de.png) 

Please try debug at this point and let me know result, because we cannot accept random bugs :) thanks for collaboration :+1:

**Comment by geerlingguy** [comment]
I get this error here and there when doing updates; note that my static content is in an NFS filesystem in production, so it is a little slow to read/write during static content deploy and setup:update. Just putting this out there, as I see it pop up in logs here and there, for no rhyme or reason. Usually the commands where I see it appear still work fine and complete though.

**Comment by ipascual** [comment]
Same issue happens randomly on Magento Cloud 

`[RuntimeException]
Error while waiting for package deployed: 19104; Status: 0`

which I know they have a network hard drive for pub/static

**Comment by ihor-sviziev** [comment]
Hi @engcom-backlog-nazar,
I have this issue too.

Did you tried to reproduce this issue inside docker on image php:7.1?

**Comment by ghost** [comment]
Hi @ihor-sviziev with php7.1 and without docker. i thinks this may be related to environment in cloud, did you have this issue on local environment ?

**Comment by CajuCLC** [comment]
This might be caused by this: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Deploy/Process/Queue.php#L391

And the function for the timeout: https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Deploy/Process/Queue.php#L372

And that all goes to const `DEFAULT_MAX_EXEC_TIME = 400;`
https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Deploy/Process/Queue.php#L32

So it looks like if the static code deploy runs for over 400 seconds, it will fail. Maybe increase this time or create an option in the admin where this can be set. If you are using NFS it might be slow to generated all themes or even some heavy themes. And it looks like that's what it was done to ECE: https://github.com/magento/ece-tools/pull/418
We are using EFS and sometimes it fails, sometimes it doesn't.

**Comment by magento-engcom-team** [comment]
Hi @talset. Thank you for your report. 
The issue has been fixed in magento/magento2#22607 by @davidalger in 2.3-develop branch
Related commit(s):
 - [49b3c7213d6a968f478d479e2063855573acd880](https://github.com/magento/magento2/commit/49b3c7213d6a968f478d479e2063855573acd880)
 - [7631cbc6d0323e2a5feeda76c52ecfb53c0be723](https://github.com/magento/magento2/commit/7631cbc6d0323e2a5feeda76c52ecfb53c0be723)
 - [e1525a80036e89c7289e22530067db522441c555](https://github.com/magento/magento2/commit/e1525a80036e89c7289e22530067db522441c555)
 - [8d0a3d8acf11807773a326f4dc03ca8d9704dc77](https://github.com/magento/magento2/commit/8d0a3d8acf11807773a326f4dc03ca8d9704dc77)
 - [2be310574b2b8bb4832300af4a50e448ff19da90](https://github.com/magento/magento2/commit/2be310574b2b8bb4832300af4a50e448ff19da90)

The fix will be available with the upcoming 2.3.3 release.

**Comment by sidolov** [comment]
Hi @talset. Thank you for your report. 
The issue has been fixed in magento/magento2#22610 by @davidalger in 2.2-develop branch
Related commit(s):
 - [fddb9f6fe752e178645a1e468b435a1ef47af663](https://github.com/magento/magento2/commit/fddb9f6fe752e178645a1e468b435a1ef47af663)
 - [158f9a4aabc4072e52a583e7d1f5ebf8b75afa9c](https://github.com/magento/magento2/commit/158f9a4aabc4072e52a583e7d1f5ebf8b75afa9c)
 - [db729ab62a152909404be93699009951e16f65f4](https://github.com/magento/magento2/commit/db729ab62a152909404be93699009951e16f65f4)
 - [ba771cfbde148970f1d48ef7424f4cdcc7f75d5b](https://github.com/magento/magento2/commit/ba771cfbde148970f1d48ef7424f4cdcc7f75d5b)

The fix will be available with the upcoming 2.2.10 release.
